### PR TITLE
[4.0] convert to prepared statements MenuType table

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
converted sql to prepared statements

### Testing Instructions

- Create a Menu
- Delete a Menu
- Update a Menu

### Expected result
works as before

### Actual result
N/A

### CODE REVIEW

File: libraries/src/Table/MenuType.php [hunk 1]
```diff
@@ -190,18 +204,28 @@ public function delete($pk = null)
...
-				->select('id')
-				->from('#__menu')
-				->where('menutype=' . $this->_db->quote($table->menutype))
-				->where('(checked_out NOT IN (0,' . (int) $userId . ') OR home=1 AND language=' . $this->_db->quote('*') . ')');
+				->select($this->_db->quoteName('id'))
+				->from($this->_db->quoteName('#__menu'))
+				->where($this->_db->quoteName('menutype') . ' = :menutype')
+				->where('(' .
+					$this->_db->quoteName('checked_out') . ' NOT IN (0, :id)' .
+					' OR ' . $this->_db->quoteName('home') . ' =1 ' .
```
Comment (Quy) [inline libraries/src/Table/MenuType.php]:
```suggestion
					' OR ' . $this->_db->quoteName('home') . ' = 1' .
```

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 060335a946a1967f90d36a268deeef86b0892672<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28565">issues.joomla.org/tracker/joomla-cms/28565</a>.</sub>

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 060335a946a1967f90d36a268deeef86b0892672<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28565">issues.joomla.org/tracker/joomla-cms/28565</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28565">issues.joomla.org/tracker/joomla-cms/28565</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**